### PR TITLE
feat: add observability provider controller

### PR DESCRIPTION
**Pull request by floreks**
- Fixed an issue with `StackDefinition` deletion
- Added `ObservabilityProvider` controller
  - Depending on the defined type it requires secret reference with special key(s). I have added a comment
    with information about the required key names in the CRD and regenerated CRD docs.
- Updated `InfrastructureStack` controller to support `ObservabilityMetric` list
  - It requires and `Identifier` and a object reference to the `ObservabilityProvider`

### test.yaml
```yaml
apiVersion: v1
kind: Namespace
metadata:
  name: floreks-stack-basic
---
apiVersion: v1
kind: Secret
metadata:
  name: floreks-datadog-provider-credentials
  namespace: floreks-stack-basic
stringData:
  apiKey: "test"
  appKey: "test"
---
apiVersion: deployments.plural.sh/v1alpha1
kind: ObservabilityProvider
metadata:
  name: floreks-obs-provider-basic
  namespace: floreks-stack-basic
spec:
  credentials:
    datadog:
      name: floreks-datadog-provider-credentials
      namespace: floreks-stack-basic
  name: floreks-obs-provider-basic
  type: DATADOG
---
apiVersion: v1
kind: Secret
metadata:
  name: infrastructurestack
  namespace: floreks-stack-basic
stringData:
  your-file.json: |
    {key1: value1, key2: value2, keyN: valueN}
---
apiVersion: deployments.plural.sh/v1alpha1
kind: InfrastructureStack
metadata:
  name: floreks-stack-basic
  namespace: floreks-stack-basic
spec:
  name: "floreks-stack-basic"
  detach: true
  type: TERRAFORM
  approval: true
  workdir: basic
  observableMetrics:
    - identifier: test
      observabilityProviderRef:
        name: floreks-obs-provider-basic
        namespace: floreks-stack-basic
  configuration:
    image: ghcr.io/pluralsh/harness
    version: 0.4.41-terraform-1.8.2
  repositoryRef:
    name: tf-test
    namespace: default
  clusterRef:
    name: existing
    namespace: default
  git:
    ref: main
    folder: terraform
  files:
    - mountPath: .
      secretRef:
        name: infrastructurestack
  environment:
    - name: TF_VAR_TEST
      value: test
```

## Test Plan

Locally and on `plrl-dev-aws` env.

## Checklist

- [ ] If required, I have updated the Plural documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] I have added a meaningful title and summary to convey the impact of this PR to a user.

### CODE REVIEW

File: go/controller/api/v1alpha1/observabilityprovider_types.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+	// +kubebuilder:validation:Required
+	Name string `json:"name"`
+
+	// Type of the ObservabilityProvider.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Enum=DATADOG;NEWRELIC
+	Type client.ObservabilityProviderType `json:"type"`
+
+	// Credentials to access the configured provider Type.
+	// +kubebuilder:validation:Required
```
Comment (michaeljguarino) [inline go/controller/api/v1alpha1/observabilityprovider_types.go:74]: These shouldn't be required ideally (If you already created in-ui)

**Review by michaeljguarino — COMMENTED** [review]
Does this support read-only mode?  It's probably ok to just ship w/out to expedite demo but we should add it in if possible

**Review by michaeljguarino — APPROVED** [review]

**Comment by michaeljguarino** [comment]
Will add a ticket to add read-only mode

File: go/controller/api/v1alpha1/observabilityprovider_types.go (hunk 1 above)
Comment (floreks) [reply go/controller/api/v1alpha1/observabilityprovider_types.go:74]: IIRC GraphQL schema required it, that's why I have made it also required.

**Comment by floreks** [comment]
Ye, readonly mode is currently not supported. Should be simple to change.

**Pull request merged**